src/app/components/algorithms/algorithm-list/algorithm-list.component.ts

```diff
@@ -38,84 +44,140 @@ export class AlgorithmListComponent implements OnInit {
       size: 10,
       totalElements: 2,
       totalPages: 5,
-      number: 1,
+      number: 0,
     },
   };
   paginatorConfig = {
-    amountChoices: [10, 20, 30],
-    selectedAmount: 10,
+    amountChoices: [1, 2, 3],
+    selectedAmount: 1,
   };
-  constructor(private algorithmService: AlgorithmService) {}
+
+  constructor(
+    private algorithmService: AlgorithmService,
+    private genericDataService: GenericDataService,
+    private dialog: MatDialog,
+    private router: Router
+  ) {}
 
   ngOnInit(): void {
-    this.algorithms = [
-      {
-        id: '261642e3-da41-4c91-b1d9-b9ce40b698a9',
-        name: 'Alg1',
-        description: 'Alg 1 description',
-        authors: [
-          'Author Ab',
-          'Author Bc',
-          'Author De',
-          'Author Ef',
-          'Author Fg',
-        ],
-        format: 'String',
-      },
-      {
-        id: '261642e3-da41-4c91-b1d9-b9ce40b698a7',
-        name: 'Alg2',
-        description: 'Alg 2 description',
-        authors: ['Author Ab', 'Author Bc'],
-        format: 'Integer',
-      },
-    ];
-    this.getAlgorithms();
+    this.getAlgorithms(this.generateGetParams());
   }
 
-  getAlgorithms(): void {
-    // TODO: Fix generated services
-    /* this.algorithmService.getAlgorithms().subscribe((data) => {
-      const algorithms = data._embedded.algorithmDtoes;
-      const page = data.page;
-      console.log(algorithms);
-      console.log(page);
-    }); */
+  getAlgorithms(params: any): void {
+    this.algorithmService.getAlgorithms(params).subscribe((data) => {
+      this.prepareAlgorithmdata(JSON.parse(JSON.stringify(data)));
+    });
   }
 
-  adjustInput(): void {}
+  getAlgorithmsHateoas(url: string): void {
+    this.genericDataService.getData(url).subscribe((data) => {
+      this.prepareAlgorithmdata(data);
+    });
+  }
+
+  prepareAlgorithmdata(data): void {
+    // Clear current algorithms
+    this.algorithms = [];
+
+    // Read all incoming algorithms
+    const classicAlgs = data._embedded.classicAlgorithmDtoes;
```

Comment:
This won't be necessary anymore.
Algorithms now end up in `_embedded.algorithms`, regardless of type.
see PlanQK/qc-atlas#99

Reply:
We will fix then after we merge the master, this was our workaround to test the backend